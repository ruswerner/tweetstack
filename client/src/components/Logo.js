import React, {PureComponent} from 'react';
import IconButton from 'material-ui/IconButton';
import AVEqualizerIcon from 'material-ui/svg-icons/av/equalizer';

export default class Logo extends PureComponent {
  render() {
    return (
      <IconButton disableTouchRipple={true}
                  style={{transform: 'rotate(180deg)'}}
                  onClick={() => window.open('https://github.com/ruswerner/tweetstack')}>
        <AVEqualizerIcon color="white"/>
      </IconButton>
    );
  }
}
