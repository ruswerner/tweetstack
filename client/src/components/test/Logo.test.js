import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Logo from '../Logo';
import renderer from 'react-test-renderer';

describe('Logo', function() {

  it('renders the logo', () => {
    const tree = renderer.create(<MuiThemeProvider><Logo/></MuiThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
