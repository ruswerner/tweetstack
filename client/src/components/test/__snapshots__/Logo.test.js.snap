// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders the logo 1`] = `
<button
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOut={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
      "background": "none",
      "border": 10,
      "boxSizing": "border-box",
      "cursor": "pointer",
      "display": "inline-block",
      "fontFamily": "Roboto, sans-serif",
      "fontSize": 0,
      "fontWeight": "inherit",
      "height": 48,
      "margin": 0,
      "muiPrepared": true,
      "outline": "none",
      "overflow": "visible",
      "padding": 12,
      "position": "relative",
      "textDecoration": "none",
      "transform": "rotate(180deg)",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "verticalAlign": null,
      "width": 48,
    }
  }
  tabIndex={0}
  type="button"
>
  <svg
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fill": "white",
        "height": 24,
        "muiPrepared": true,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "userSelect": "none",
        "width": 24,
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M10 20h4V4h-4v16zm-6 0h4v-8H4v8zM16 9v11h4V9h-4z"
    />
  </svg>
</button>
`;
